src/SLCore/Analysis/SLCoreRuleSettings.cs

```diff
@@ -0,0 +1,81 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.UserRuleSettings;
+using SonarLint.VisualStudio.SLCore.Core;
+using SonarLint.VisualStudio.SLCore.Service.Rules;
+using SonarLint.VisualStudio.SLCore.Service.Rules.Models;
+
+namespace SonarLint.VisualStudio.SLCore.Analysis;
+
+public interface ISLCoreRuleSettings
+{
+    Dictionary<string, StandaloneRuleConfigDto> RulesSettings { get; }
+    void UpdateStandaloneRulesConfiguration();
```

Comment:
It feels like these 2 things don't really belong together. We have UserSettingsUpdater and UserSettingsProvider split, but SLCoreSettings provider and updater are in the same class together

Reply:
Yes, because the UserSettingsUpdater, which is in the Integration.Vsix assembly, is responsible to update the all the RuleSettings. 

UserSettingsProvider, which is in the Core assembly takes care of the settings.json file.

Finally, the SlCoreSettings takes care of sending the settings to SlCore. So I think it makes sense that the settings and sending the update to SlCore should be together. Would be more confusing for the consumers if I split the logic into multiple classes.

Reply:
For example, `SLCoreInstanceHandle` would never need to access `UpdateStandaloneRulesConfiguration`, so, technically, the interface is not minimal. To me it still feels like a provider and a updater got merged together.

Reply:
Will be done as part of the architecture refactoring, in https://sonarsource.atlassian.net/browse/SLVS-1301